src/main/groovy/org/edx/jenkins/dsl/AnalyticsConstants.groovy

```diff
@@ -0,0 +1,114 @@
+            textParam('EMR_YARN_SITE_CONFIG', allVars.get('EMR_YARN_SITE_CONFIG'), 'EMR yarn-site configuration.')
+            stringParam('EMR_APPLICATIONS_CONFIG', allVars.get('EMR_APPLICATIONS_CONFIG'), 'Applications to install on EMR cluster.')
+            textParam('EXTRA_VARS', allVars.get('EMR_EXTRA_VARS'), $/Extra variables to pass to the EMR provision/terminate ansible playbook.
+This text may reference other parameters in the task as shell variables, e.g.  $$CLUSTER_NAME./$)
+            stringParam('NOTIFY', '$PAGER_NOTIFY', 'Space separated list of emails to send notifications to.')
+            stringParam('NUM_TASK_CAPACITY', env.get('NUM_TASK_CAPACITY', allVars.get('NUM_TASK_CAPACITY')), 'Number of EMR instance capacity to use for this job.')
+            stringParam('SECURE_BRANCH', '$ANALYTICS_SECURE_RELEASE', 'e.g. tagname or origin/branchname, or $ANALYTICS_SECURE_RELEASE when released.')
+            stringParam('SECURE_CONFIG', env.get('SECURE_CONFIG', 'analytics-tasks/prod-edx.cfg'), '')
+            stringParam('SECURE_REPO', allVars.get('SECURE_REPO_URL'), '')
+            stringParam('TASKS_BRANCH', '$ANALYTICS_PIPELINE_RELEASE', 'e.g. tagname or origin/branchname,  e.g. origin/master or $ANALYTICS_PIPELINE_RELEASE')
+            stringParam('TASKS_REPO', 'https://github.com/edx/edx-analytics-pipeline.git', 'Git repo containing the analytics pipeline tasks.')
+            stringParam('TASK_USER', allVars.get('TASK_USER'), 'User which runs the analytics task on the EMR cluster.')
+            booleanParam('TERMINATE', true, 'Terminate the EMR cluster after running the analytics task?')
+        }
+    }
+
+    public static def date_interval_parameters = { allVars ->
+      return {
+        stringParam('FROM_DATE', allVars.get('FROM_DATE', '2013-11-01'),
+          'The first date to export data for. Data for this date and all days before the "TO_DATE" parameter' +
+          ' will be included.' + $/
+  Format: A string that can be parsed by the GNU coreutils "date" utility.
+
+  Examples:
+  * today
+  * yesterday
+  * 3 days ago
+  * 1 week ago
+  * 2013-11-01
+  /$)
+        stringParam('TO_DATE', allVars.get('TO_DATE', 'today'),
+          'The day after the last date to export data for. Data from the "FROM_DATE" parameter to 11:59:59' +
+          ' on the date before this date will be included.' + $/
+  Format: A string that can be parsed by the GNU coreutils "date" utility.
+
+  Examples:
+  * today
+  * yesterday
+  * 3 days ago
+  * 1 week ago
+  * 2017-11-01
+  /$)
+      }
+    }
+
+    public static def common_wrappers = { allVars ->
+        return {
+            sshAgent('1')
+            timestamps()
+        }
+    }
+
+    public static def common_publishers = { allVars ->
+        return {
+            mailer('$NOTIFY')
+        }
+    }
+
+    public static def common_triggers = { allVars, env=[:] ->
```

Comment:
Nit: This might end up being more confusing in the long run.  If someone has to quickly check a job, and they see `common_triggers` in the job DSL, they won't know if the job is scheduled or not, without digging further. What do you think about making the job explicitly add a cron statement, and then only add the closure to the DSL if you want to add the trigger?

Reply:
Do you mean something like this in the Job ?
```groovy
job_frequency = extraVars.get('JOB_FREQUENCY')
if (job_frequency) {
   triggers common_triggers(job_frequency)
}
```

Reply:
@estute, it was done this way because each environment specified in job config uses a different trigger frequency.  It really just boils down to whether that data is stored in analytics-secure vs. jenkins-job-dsl in the job DSL groovy file.  I'm okay with either option.

Reply:
Understood. If it makes more sense for the team using these jobs to check analytics-secure for configurations, that's fine by me.